diff --git a/python/setup.py b/python/setup.py
--- a/python/setup.py
+++ b/python/setup.py
@@ -704,10 +704,10 @@
     entry_points={
         "console_scripts": [
             "ray=ray.scripts.scripts:main",
-            "rllib=ray.rllib.scripts:cli [rllib]",
-            "tune=ray.tune.cli.scripts:cli",
-            "ray-operator=ray.ray_operator.operator:main",
-            "serve=ray.serve.scripts:cli",
+            #"rllib=ray.rllib.scripts:cli [rllib]",
+            #"tune=ray.tune.cli.scripts:cli",
+            #"ray-operator=ray.ray_operator.operator:main",
+            #"serve=ray.serve.scripts:cli",
         ]
     },
     include_package_data=True,
-- 
2.30.1 (Apple Git-130)
