diff --git a/bazel/ray_deps_setup.bzl b/bazel/ray_deps_setup.bzl
index ce5bd7f23a..2fa0e5b7ca 100644
--- a/bazel/ray_deps_setup.bzl
+++ b/bazel/ray_deps_setup.bzl
@@ -99,8 +99,8 @@ def ray_deps_setup():
         name = "com_github_antirez_redis",
         build_file = "@com_github_ray_project_ray//bazel:BUILD.redis",
         patch_args = ["-p1"],
-        url = "https://github.com/redis/redis/archive/refs/tags/7.0.8.tar.gz",
-        sha256 = "0e439cbc19f6db5a4c63d355519ab73bf6ac2ecd47df806c14b19564b3d0c593",
+        url = "https://github.com/redis/redis/archive/refs/tags/7.0.12.tar.gz",
+        sha256 = "13d4689454e29e7b9f1161b544e6d08b0ddd27d057859fde7b1916869b3bf701",
         patches = [
             "@com_github_ray_project_ray//thirdparty/patches:redis-quiet.patch",
         ],
@@ -147,8 +147,8 @@ def ray_deps_setup():
     auto_http_archive(
         name = "bazel_skylib",
         strip_prefix = None,
-        url = "https://github.com/bazelbuild/bazel-skylib/releases/download/1.0.2/bazel-skylib-1.0.2.tar.gz",
-        sha256 = "97e70364e9249702246c0e9444bccdc4b847bed1eb03c5a3ece4f83dfe6abc44",
+        url = "https://github.com/bazelbuild/bazel-skylib/releases/download/1.3.0/bazel-skylib-1.3.0.tar.gz",
+        sha256 = "74d544d96f4a5bb630d465ca8bbcfe231e3594e5aae57e1edbf17a6eb3ca2506",
     )

     auto_http_archive(

