From 13b0128613727f7034864d80470a42876a8021ae Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Fri, 27 Feb 2026 02:29:58 -0800
Subject: [PATCH] fix redis build on macos

---
 bazel/redis.BUILD | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/bazel/redis.BUILD b/bazel/redis.BUILD
index fd1cc06..649281e 100644
--- a/bazel/redis.BUILD
+++ b/bazel/redis.BUILD
@@ -46,17 +46,17 @@ make(
 
 genrule_cmd = select({
     "@platforms//os:osx": """
-        export CC=$(CC)
+        export CC=$${CC_FOR_BUILD}
         export CFLAGS=$(CC_FLAGS)
-        export AR=$${CC/gnu-cc/gnu-ar}
-        export NM=$${CC/gnu-cc/gnu-nm}
-        export RANLIB=$${CC/gnu-cc/gnu-ranlib}
+        export AR=$${AR/gnu-cc/gnu-ar}
+        export NM=$${NM/gnu-cc/gnu-nm}
+        export RANLIB=$${RANLIB/gnu-cc/gnu-ranlib}
         tmpdir="redis.tmp"
         p=$(location Makefile)
         cp -p -L -R -- "$${p%/*}" "$${tmpdir}"
         chmod +x "$${tmpdir}"/deps/jemalloc/configure
         parallel="$$(getconf _NPROCESSORS_ONLN || echo 1)"
-        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body" AR="$${AR}" RANLIB="$${RANLIB}"
+        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body" AR="$${AR}" RANLIB="$${RANLIB} -isysroot $${CONDA_BUILD_SYSROOT} -mmacosx-version-min=$${macos_min_version}" LDFLAGS="$${LDFLAGS-} -mmacosx-version-min=$${macos_min_version}"
         mv "$${tmpdir}"/src/redis-server $(location redis-server)
         chmod +x $(location redis-server)
         mv "$${tmpdir}"/src/redis-cli $(location redis-cli)
-- 
2.39.3 (Apple Git-146)

