From 8d9bbf5e0b1156d2880fc57bdff191929a401701 Mon Sep 17 00:00:00 2001
From: Jean-Christophe Morin <jcmorin@anaconda.com>
Date: Wed, 4 Oct 2023 17:29:42 -0400
Subject: [PATCH] Fix bazel build of redis on macOS when -isysroot is used.

---
 bazel/BUILD.redis | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bazel/BUILD.redis b/bazel/BUILD.redis
index 60e070028f..dd5dddd3f2 100644
--- a/bazel/BUILD.redis
+++ b/bazel/BUILD.redis
@@ -42,17 +42,17 @@ make(
 
 genrule_cmd = select({
     "@bazel_tools//src/conditions:darwin": """
-        export CC=$(CC)
+        export CC=$${CC_FOR_BUILD}
         export CFLAGS=$(CC_FLAGS)
-        export AR=$${CC/gnu-cc/gnu-ar}
-        export NM=$${CC/gnu-cc/gnu-nm}
-        export RANLIB=$${CC/gnu-cc/gnu-ranlib}
+        export AR=$${AR/gnu-cc/gnu-ar}
+        export NM=$${NM/gnu-cc/gnu-nm}
+        export RANLIB=$${RANLIB/gnu-cc/gnu-ranlib}
         tmpdir="redis.tmp"
         p=$(location Makefile)
         cp -p -L -R -- "$${p%/*}" "$${tmpdir}"
         chmod +x "$${tmpdir}"/deps/jemalloc/configure
         parallel="$$(getconf _NPROCESSORS_ONLN || echo 1)"
-        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body" AR="$${AR}" RANLIB="$${RANLIB}"
+        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body" AR="$${AR}" RANLIB="$${RANLIB} -isysroot $${CONDA_BUILD_SYSROOT} -mmacosx-version-min=$${macos_min_version}" LDFLAGS="$${LDFLAGS-} -mmacosx-version-min=$${macos_min_version}"
         mv "$${tmpdir}"/src/redis-server $(location redis-server)
         chmod +x $(location redis-server)
         mv "$${tmpdir}"/src/redis-cli $(location redis-cli)
-- 
2.42.0

