From 11685ce59fe2a5fa0d8f1ded7b6e21a38563b6e1 Mon Sep 17 00:00:00 2001
From: Mohamed Sentissi <msentissi@anaconda.com>
Date: Fri, 8 Nov 2024 17:46:45 -0500
Subject: [PATCH] ignore stringop-overflow on linux

---
 .bazelrc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.bazelrc b/.bazelrc
index 7a6b2df..5e2f1bb 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -52,7 +52,8 @@ build --per_file_copt="\\.pb\\.cc$@-w"
 # Ignore one specific warning
 build --per_file_copt="event\\.cc$@-w"
 build --per_file_copt="logging\\.cc$@-w"
+build:linux --per_file_copt="external/upb/upbc/protoc-gen-upbdefs.cc@-Wno-error=implicit-function-declaration"
 build:linux --per_file_copt="-\\.(asm|S)$,external/.*@-w,-Wno-error=implicit-function-declaration"
 build:macos --per_file_copt="-\\.(asm|S)$,external/.*@-w,-Wno-error=implicit-function-declaration"
 # Ignore warnings for host tools, which we generally can't control.
 # Ideally we only want to ignore warnings for external project
-- 
2.45.2

