From 537269a0587eb1b52ec3e0c9e031cd581e3c2850 Mon Sep 17 00:00:00 2001
From: Mohamed Sentissi <msentissi@anaconda.com>
Date: Fri, 8 Nov 2024 16:38:13 -0500
Subject: [PATCH] downgrade FORTIFY_SOURCE in aarch64

---
 bazel/BUILD.redis | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bazel/BUILD.redis b/bazel/BUILD.redis
index 4adfd13..c64da33 100644
--- a/bazel/BUILD.redis
+++ b/bazel/BUILD.redis
@@ -28,6 +28,7 @@ make(
         "-Wno-pragmas",
         "-Wno-empty-body",
         "-fPIC",
+        "-D_FORTIFY_SOURCE=1",
     ],
     visibility = ["//visibility:public"],
     lib_source = ":all_srcs",
-- 
2.45.2

